src/renderer/families/cosmos/DelegationFlowModal/fields/ValidatorsField.js

```diff
@@ -0,0 +1,246 @@
+
+/** @TODO move this in common */
+/** Search filters for validator list */
+const searchFilter = (query?: string) => ({
+  name,
+  address,
+}: {
+  name: ?string,
+  address: string,
+}) => {
+  if (!query) return true;
+  const terms = `${name || ""} ${address}`;
+  return terms.toLowerCase().includes(query.toLowerCase().trim());
+};
+
+/** @TODO move this in common */
+/** Hook to search and sort SR list according to initial votes and query */
+export function useSortedValidators(
+  search: string,
+  validators: CosmosValidatorItem[],
+  delegations: CosmosDelegationInfo[],
+): {
+  validator: CosmosValidatorItem,
+  name: ?string,
+  address: string,
+  rank: number,
+}[] {
+  const { current: initialVotes } = useRef(delegations.map(({ address }) => address));
+
+  const formattedValidators = useMemo(
+    () =>
+      validators.map((validator, rank) => ({
+        validator,
+        name: validator.name,
+        address: validator.validatorAddress,
+        rank: rank + 1,
+      })),
+    [validators],
+  );
+
+  const sortedVotes = useMemo(
+    () =>
+      formattedValidators
+        .filter(({ address }) => initialVotes.includes(address))
+        .concat(formattedValidators.filter(({ address }) => !initialVotes.includes(address))),
+    [formattedValidators, initialVotes],
+  );
+
+  const sr = useMemo(
+    () => (search ? formattedValidators.filter(searchFilter(search)) : sortedVotes),
+    [search, formattedValidators, sortedVotes],
+  );
+
+  return sr;
+}
+
+type Props = {
+  t: TFunction,
+  delegations: CosmosDelegationInfo[],
+  account: Account,
+  status: TransactionStatus,
+  onChangeDelegations: (updater: (CosmosDelegationInfo[]) => CosmosDelegationInfo[]) => void,
+  bridgePending: boolean,
+};
+
+const ValidatorField = ({
+  t,
+  account,
+  onChangeDelegations,
+  status,
+  bridgePending,
+  delegations,
+}: Props) => {
+  invariant(account, "cosmos account required");
+
+  const [search, setSearch] = useState("");
+  const { cosmosResources } = account;
+  invariant(cosmosResources && delegations, "cosmos transaction required");
+
+  const unit = getAccountUnit(account);
+
+  const language = useSelector(languageSelector);
+
+  const { validators } = useCosmosPreloadData();
+  const SR = useSortedValidators(search, validators, delegations);
+
+  const delegationsAvailable = formatValue(
+    cosmosResources.delegatedBalance.plus(account.spendableBalance),
+    unit,
+  );
+  const delegationsUsed = delegations.reduce((sum, v) => sum + formatValue(v.amount, unit), 0);
+  const delegationsSelected = delegations.length;
+
+  const max = Math.max(0, delegationsAvailable - delegationsUsed);
+
+  const onUpdateDelegation = useCallback(
+    (address, value) => {
+      const raw = value ? parseInt(value.replace(/[^0-9]/g, ""), 10) : 0;
+
+      const amount =
+        raw <= 0 || delegationsSelected > MAX_VOTES
+          ? BigNumber(0)
+          : BigNumber(raw).multipliedBy(10 ** unit.magnitude);
```

Comment:
raw should be already in correct magnitude in the first place if we use a AmountInput component. (we should try to use it / adapt from it what needs to be adapted)